### ISSUE TITLE
DataDir/Checkpoint file configuration is broken

### ISSUE THREAD

**Issue by isobit** [aws/amazon-ecs-agent#2850]
### Summary
Checkpoints are not enabled when configuring using a file (e.g. `/etc/ecs_container_agent/config.json`), even if `DataDir` is defined and `Checkpoint` is `true` due to new config merging logic introduced in 17e5493a0592b9e4e18c2b276cbe27ad09d89547 and 481a2487c86d9ef6f9a0b3f92a9be7a886db1767.

### Description

When the config is loaded [here](https://github.com/aws/amazon-ecs-agent/blob/5c82161077b4045b2b64d29e8555f2522aff8419/agent/config/config.go#L221-L257), the environment config is constructed first and then the file config is merged in.

When the environment config is read, it calls `parseCheckpoint` [here](https://github.com/aws/amazon-ecs-agent/blob/5c82161077b4045b2b64d29e8555f2522aff8419/agent/config/config.go#L522). Because we are only passing config via a file (no env vars), `parseCheckpoint` will [return](https://github.com/aws/amazon-ecs-agent/blob/5c82161077b4045b2b64d29e8555f2522aff8419/agent/config/parse.go#L41) a value of `ExplicitlyDisabled`.

In previous versions, `parseCheckpoint` would return `false`, which would be [treated as a `ZeroOrNil` value](https://github.com/aws/amazon-ecs-agent/blob/a250409cf5eb4ad84a7b889023f1e4d2e274b7ab/agent/config/config.go#L197-L199) during merging, meaning the `Checkpoint` value from the file config would take precedence. However, in 17e5493a0592b9e4e18c2b276cbe27ad09d89547 and 481a2487c86d9ef6f9a0b3f92a9be7a886db1767 this was changed such that `parseCheckpoint` now [returns `ExplicitlyDisabled`](https://github.com/aws/amazon-ecs-agent/blob/5c82161077b4045b2b64d29e8555f2522aff8419/agent/config/parse.go#L41) (of type `BooleanDefaultFalse`), and `Merge` now has [special logic for `BooleanDefaultFalse`](https://github.com/aws/amazon-ecs-agent/blob/5c82161077b4045b2b64d29e8555f2522aff8419/agent/config/config.go#L201-L205) which only lets the right value take precedence if the JSON-marshaled left value is `null`, which is not the case with `ExplicitlyDisabled`, since it JSON-marshals to `false`.

When no env vars are set, `environmentConfig()` returns `Checkpoint: ExplicitlyDisabled`, which means the `Checkpoint` returned by `fileConfig()` can never take precedence during a merge, and so the file config values are ignored.

### Expected Behavior
Checkpoints are enabled with the following config in `/etc/ecs_container_agent/config.json`:

```json
{
  "DataDir": "/var/run/ecs_agent",
  "Checkpoint": true
}
```

### Observed Behavior
The agent does not enable checkpointing and logs the following:

```Checkpointing not enabled; a new container instance will be created each time the agent is run```

Checkpointing _is_ enabled when `ECS_DATADIR=/var/run/ecs_agent` is passed in the environment (we're using this as a workaround).

### Environment Details
ECS Agent version: 1.51.0

**Comment by fierlion** [comment]
I'm working on repro now -- thanks for the detailed overview!

**Comment by mythri-garaga** [comment]
Closing this issue as it is fixed in [v1.53.1](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.53.1).
